core/src/test/java/feast/core/service/SpecServiceTest.java

```diff
@@ -706,6 +705,117 @@ public void applyFeatureSetShouldUpdateFeatureSetWhenConstraintsAreUpdated()
     }
   }
 
+  @Test
+  public void applyFeatureSetShouldAcceptLabels() throws InvalidProtocolBufferException {
+    List<EntitySpec> entitySpecs = new ArrayList<>();
+    entitySpecs.add(EntitySpec.newBuilder().setName("entity1").setValueType(Enum.INT64).build());
+
+    Map<String, String> featureLabels0 =
+        new HashMap<>() {
+          {
+            put("label1", "feast1");
+          }
+        };
+
+    Map<String, String> featureLabels1 =
+        new HashMap<>() {
+          {
+            put("label1", "feast1");
+            put("label2", "feast2");
+          }
+        };
+
+    List<Map<String, String>> featureLabels = new ArrayList<>();
+    featureLabels.add(featureLabels0);
+    featureLabels.add(featureLabels1);
+
+    List<FeatureSpec> featureSpecs = new ArrayList<>();
+    featureSpecs.add(
+        FeatureSpec.newBuilder()
+            .setName("feature1")
+            .setValueType(Enum.INT64)
+            .putAllLabels(featureLabels.get(0))
+            .build());
+    featureSpecs.add(
+        FeatureSpec.newBuilder()
+            .setName("feature2")
+            .setValueType(Enum.INT64)
+            .putAllLabels(featureLabels.get(1))
+            .build());
+
+    FeatureSetSpec featureSetSpec =
+        FeatureSetSpec.newBuilder()
+            .setProject("project1")
+            .setName("featureSetWithConstraints")
```

Comment:
Unimportant but this name probably isn't relevant to this test case.

In fact though, I wish we weren't even thinking about this. The extent of repeated test setup boilerplate in Feast's code base has been pretty painful to me on internal work, to the extent that when I get a little more caught up with current code, I want to go to town with PRs adding something like [ObjectMothers with random data][1] (as much as this does not sound like a fun weekend).

Anyway, this is out of scope for now, let's carry on 😇 

[1]: https://www.jworks.io/easy-testing-with-objectmothers-and-easyrandom/